### PR TITLE
health: init at 0.94.0

### LINKED ISSUE #208915: Health - Track your fitness goals
**Project description**
Health can visualize how many steps you’ve made daily, your weight development over time and your daily activities. Data can be synched from Google Fit or manually entered into Health. Afterwards, it can be viewed and edited in Health.

**Metadata**

* homepage URL: https://gitlab.gnome.org/World/Health
* source URL: https://gitlab.gnome.org/World/Health
* license: gpl3
* platforms: linux

### PR DESCRIPTION
**Pull request by Aleksanaa**
###### Description of changes

Resolves https://github.com/NixOS/nixpkgs/issues/208915

https://gitlab.gnome.org/World/Health/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/health/default.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    patches = [ ./update_gtk4_cargo_deps.patch ];
+    name = "${pname}-${version}";
+    hash = "sha256-j0I0vKoGaf2pce2C/xkz+nJYCfLvHB5F6Q9XpJtABMI=";
+  };
+
+  patches = [
+    ./max_size_tightending_thresh.patch
+    ./password_entry_max_value.patch
```
[the remark this answers is not in the document]
Comment (Aleksanaa) [reply pkgs/applications/misc/health/default.nix:38]: The first is included in https://gitlab.gnome.org/World/Health/-/commit/9d5036d87732779eebe0674721f07265516d0ecc
The second is in https://gitlab.gnome.org/World/Health/-/commit/864d961a3c4ef30dfe290029b8abf68ee16ded19
The cargo update is mentioned in this issue https://gitlab.gnome.org/World/Health/-/issues/174
We just have to wait for the next release.

[the remark this answers is not in the document]
Comment (wegank) [reply pkgs/applications/misc/health/default.nix:38]: Or just fetchpatch from AUR...

[the remark this answers is not in the document]
Comment (Aleksanaa) [reply pkgs/applications/misc/health/default.nix:38]: You've noticed it 🤣

File: pkgs/applications/misc/health/default.nix [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    cargo
+    wrapGAppsHook4
+    blueprint-compiler
+  ];
+
+  buildInputs = [
+    libadwaita
+    libsecret
+    tracker
+  ];
```
Comment (wegank) [inline pkgs/applications/misc/health/default.nix:65]:
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [
    darwin.apple_sdk.frameworks.Security
    darwin.apple_sdk.frameworks.Foundation
  ];
```

Comment (Aleksanaa) [reply pkgs/applications/misc/health/default.nix:65]: Did you try to run it with these deps?

Comment (wegank) [reply pkgs/applications/misc/health/default.nix:65]: Capture d’écran 2023-05-10 à 08 59 39

Still building on `x86_64-darwin`.

Comment (Aleksanaa) [reply pkgs/applications/misc/health/default.nix:65]: Okay cool

**Comment by Aleksanaa** [comment]
Able to merge, I guess, if the package name wasn't too controversial

**Pull request merged**